### PR TITLE
add remote support

### PR DESCRIPTION
**Pull request by ItamarShDev**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@audyx/audyx-deps",
-  "version": "1.0.38",
+  "version": "1.0.37-remote-1",
```
Comment (rinama) [inline package.json]: Don't forget to change

File: src/cljs/audyx/app/socket.cljs [hunk 2]
```diff
@@ -8,9 +8,19 @@
...
-                  (!> js/io.connect (? js/constantsEnvironment.servicesAudyxUrl)
-                                    (clj->js {:secure true :port 443 "forceNew" true}))))
+(defonce socket
+  (let [socket (atom nil)]
+    (fn []
+      (if @socket
+        @socket
+        (when (? js/window.io)
+          (->>
+            (!> js/io.connect
```
Comment (rinama) [inline src/cljs/audyx/app/socket.cljs]: Move to function

File: src/cljs/audyx/app/socket.cljs [hunk 3]
```diff
@@ -8,9 +8,19 @@
   (! js/location.hash url)
   (js/alert txt))
 
-(defonce socket (when (? js/window.io)
-                  (!> js/io.connect (? js/constantsEnvironment.servicesAudyxUrl)
-                                    (clj->js {:secure true :port 443 "forceNew" true}))))
+(defonce socket
+  (let [socket (atom nil)]
```
Comment (rinama) [inline src/cljs/audyx/app/socket.cljs]: What is this?

File: src/cljs/audyx/remote/remote.cljs [hunk 4]
```diff
@@ -13,7 +13,7 @@
 
 (defonce data (atom {}))
 (defonce audyx-in-room (atom false))
-(defonce audyx-room (? js/window.authentication.token))
+(defn audyx-room [] (? js/window.authentication.token))
```
Comment (rinama) [inline src/cljs/audyx/remote/remote.cljs]: Move `(? js/window.authentication.token)` to new line

Comment (rinama) [inline src/cljs/audyx/remote/remote.cljs]: Add empty line above

File: src/cljs/audyx/remote/remote.cljs [hunk 5]
```diff
@@ -22,49 +22,67 @@
...
-  (disconnect)
-  (ready! false)
-  (reset! data {}))
+(deftry ^:export closeSocket []
+  (when (socket)
+    (remove-listeners)
+    (disconnect)
+    (ready! false)
+    (reset! data {})))
+
```
Comment (rinama) [inline src/cljs/audyx/remote/remote.cljs]: Remove

File: src/cljs/audyx/app/socket.cljs (hunk 3 above)
Comment (ItamarShDev) [reply src/cljs/audyx/app/socket.cljs]: function that returns the data or init it (once)

File: src/cljs/audyx/remote/remote.cljs [hunk 6]
```diff
@@ -22,49 +22,67 @@
...
                                       (create-event! :remote)))
-      (!> socket.emit "audyx::add::room" audyx-room))
+      (!> socket.emit "audyx::add::room" (audyx-room)))
     c))
 
 (defn on-data-change [_ _ previous current]
-  (when (and socket (ready?) (not= previous current))
-        (!> socket.emit "audyx::data::test" (clj->js current))))
+  (when (and (ready?) (not= previous current))
+    (when-let [socket (socket)]
```
Comment (rinama) [inline src/cljs/audyx/remote/remote.cljs]: `socket` is not used

Comment (ItamarShDev) [reply src/cljs/audyx/remote/remote.cljs]: line below...

**Pull request merged**